satellite/metainfo/metainfo.go

```diff
@@ -829,8 +781,68 @@ func getAllowedBuckets(ctx context.Context, action macaroon.Action) (_ macaroon.
 }
 
 // SetBucketAttribution sets the bucket attribution.
-func (endpoint *Endpoint) SetBucketAttribution(context.Context, *pb.BucketSetAttributionRequest) (resp *pb.BucketSetAttributionResponse, err error) {
-	return resp, status.Error(codes.Unimplemented, "not implemented")
+func (endpoint *Endpoint) SetBucketAttribution(ctx context.Context, req *pb.BucketSetAttributionRequest) (resp *pb.BucketSetAttributionResponse, err error) {
+	defer mon.Task()(&ctx)(&err)
+
+	err = endpoint.setBucketAttribution(ctx, req.Name, req.AttributionId)
+
+	return &pb.BucketSetAttributionResponse{}, err
+}
+
+func (endpoint *Endpoint) setBucketAttribution(ctx context.Context, bucketName []byte, parterID []byte) error {
```

Comment:
curious why have this 'setBucketAttribution'. I believe the other function 'Old' will be removed. So can we make one function with 'SetBucketAttribution`.

Reply:
It will be removed but I'm not sure when. For now we need to keep `Old` version to be backward compatible.